### PR TITLE
Signalr keys

### LINKED ISSUE #3406: Feature: expose SignalR credentials
Feature: expose SignalR credentials: keys and connection strings 

Azure Service Operator version: 2
apiVersion: signalrservice.azure.com/v1api20211001
kind: SignalR

Please consider adding operatorSpec to the SignalR
```
  operatorSpec:
    secrets:
      key1:
        name: signalrsecret
        key: key1
      key2:
        name:  signalrsecret
        key: key2
      ConnectionString1:
        name: signalrsecret
        key: ConnectionString1
      ConnectionString2:
        name:  signalrsecret
        key: ConnectionString2
```

### PR DESCRIPTION
**Pull request by mehighlow**
Closes #3406

**What this PR does / why we need it**:
This PR adds SignalR API keys export

**If applicable**:
- [ ] this PR contains documentation
- [x] this PR contains tests
- [x] this PR contains YAML Samples

### CODE REVIEW

File: v2/api/signalrservice/customizations/signal_r_extention_authorization.go [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+	// Using armClient.ClientOptions() here ensures we share the same HTTP connection, so this is not opening a new
+	// connection each time through
+	clientFactory, err := armsignalr.NewClientFactory(id.SubscriptionID, armClient.Creds(), armClient.ClientOptions())
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to create ARM signalR client factory")
+	}
+
+	res, err := clientFactory.NewClient().ListKeys(ctx, id.ResourceGroupName, typedObj.AzureName(), nil)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to finish the request")
```
Comment (matthchr) [inline v2/api/signalrservice/customizations/signal_r_extention_authorization.go:67]:
```suggestion
		return nil, errors.Wrapf(err, "failed to list keys")
```

File: v2/internal/controllers/crd_signalr_test.go [hunk 2]
```diff
@@ -97,6 +100,27 @@ func Test_SignalRService_SignalR_CRUD(t *testing.T) {
 	tc.Expect(signalR.Status.Cors).ToNot(BeNil())
 	tc.Expect(signalR.Status.Cors.AllowedOrigins).To(ContainElement("https://definitelymydomain.horse"))
 
+	// There should be no secrets at this point
+	list := &v1.SecretList{}
+	tc.ListResources(list, client.InNamespace(tc.Namespace))
+	tc.Expect(list.Items).To(HaveLen(0))
+
+	// Run sub-tests on the cosmosdb in sequence
```
Comment (matthchr) [inline v2/internal/controllers/crd_signalr_test.go:108]:
```suggestion
	// Run sub-tests on the SignalR service in sequence
```

File: v2/internal/controllers/crd_signalr_test.go [hunk 3]
```diff
@@ -105,3 +129,68 @@ func Test_SignalRService_SignalR_CRUD(t *testing.T) {
...
+}
+
+func SignalR_SecretsWrittenToDifferentKubeSecrets(tc *testcommon.KubePerTestContext, signalR *signalrservice.SignalR) {
+	old := signalR.DeepCopy()
+	primaryKeySecret := "secret1"
+	primaryConnectionString := "secret2"
+	secondaryKeySecret := "secret3"
+	secondaryConnectionString := "secret4"
+
+	// Not testing port as it's not returned by default so won't be written anyway
```
Comment (matthchr) [inline v2/internal/controllers/crd_signalr_test.go:168]: I think delete this comment as it doesn't apply here?

**Review by matthchr — COMMENTED** [review]
Looks great! Had a few small comments.

File: v2/internal/controllers/crd_signalr_test.go (hunk 3 above)
Comment (mehighlow) [reply v2/internal/controllers/crd_signalr_test.go]: 💯

File: go.mod [hunk 4]
```diff
@@ -37,9 +37,13 @@ require (
 )
 
 require (
+	github.com/Azure/azure-sdk-for-go/sdk/azcore v1.6.0 // indirect
```
[the remark this answers is not in the document]
Comment (mehighlow) [reply go.mod]: Mistakenly ran go mod tidy in root folder. Reverted both `go.mod` and `go.sum`

**Pull request merged**